File: 0001_0100/0093/0093.py
def comp(s, t):
    n = len(s)
    k = sum(1 for i in range(n) if s[i] != t[i]) if len(t) == n else 0
    return 1 if k == 1 else 0            

with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s, t = [], []
    for i in range(int(f.readline())):
        s.append(f.readline().strip())
    for i in range(int(f.readline())):
        t.append(f.readline().strip())
    q.write(' '.join([str(sum(comp(a, b) for b in t)) for a in s]))
        

File: 0001_0100/0093/0093.c
#include <stdio.h>
#include <string.h>

int comp(char *s, char *t)
{
	int k = 0, n = strlen(s);
	if (n != strlen(t))
		return 0;
	for (int i = 0; i < n && k < 2; ++i)
		k += (s[i] != t[i]);
	return k == 1 ? 1 : 0;
}

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, m, i, j;
	char s[30][31], t[30][31];

	fscanf(f, "%d", &n);
	for (i = 0; i < n; ++i)
		fscanf(f, "%s", s[i]);
	fscanf(f, "%d", &m);
	for (i = 0; i < m; ++i)
		fscanf(f, "%s", t[i]);
	for (i = 0; i < n; ++i)
	{
		int p = 0;
		for (j = 0; j < m; ++j)
			p += comp(s[i], t[j]);
		fprintf(q, "%d ", p);
	}
	return 0;
}
